databroker/broker.py

```diff
@@ -708,6 +708,62 @@ def process(self, headers, func, fields=None, fill=False):
 
     get_fields = staticmethod(get_fields)  # for convenience
 
+    def export(self, headers, db, new_root=None):
+        """
+        Export a list of headers.
+
+        .. warning::
+
+            This does not yet support copying external files.
+
+        Parameters:
+        -----------
+        headers : databroker.header
+            one or more headers that are going to be exported
+        db : databroker.Broker
+            an instance of databroker.Broker class, which has
+            filestore (fs) and metadatastore (mds) attributes
+            that will be the target to export info
+        new_root : str
+            optional. root directory of files that are going to
+            be exported
+        """
+        try:
+            headers.items()
+        except AttributeError:
+            pass
+        else:
+            headers = [headers]
+        for header in headers:
+            # insert mds
+            db.mds.insert_run_start(**_clean_doct(header['start']))
+            events = self.get_events(header)
+            for descriptor in header['descriptors']:
+                db.mds.insert_descriptor(**_clean_doct(descriptor))
+                for event in events:
+                    event = _clean_doct(event)
+                    # 'filled' is obtained from the descriptor, not stored
+                    # in each event.
+                    event.pop('filled', None)
+                    db.mds.insert_event(**event)
+            db.mds.insert_run_stop(**_clean_doct(header['stop']))
+            # insert fs
+            res_uids = self.get_resource_uids(header)
+            for uid in res_uids:
+                res = self.fs.resource_given_uid(uid)
+                db.fs.insert_resource(res['spec'],
+                                      res['resource_path'],
+                                      res['resource_kwargs'],
+                                      root=new_root)
+                                      # FIXME: revisit root when dealing
```

Comment:
The new_root is the root place where the external data file is going to transferred? So basically we can transfer data between different servers.